packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -332,7 +332,7 @@ class ScaffoldState extends State<Scaffold> {
         bottomSheet,
         completer,
         () => entry.remove(),
-        setState
+        (VoidCallback fn) { bottomSheetKey.currentState?.setState(fn); }
```

Comment:
There don't seem top be any uses of ScaffoldFeatureController.setState() except for the hack I added (and will remove).  However: I suppose the idea is that someone who's only holding on to the feature controller can update some state that the feature controller depends inside of myFeatureController.setState().

That's what I've done; it does work. It's not as tidy as just calling setState() once, on the application itself.